engine/access/rpc/engine.go

```diff
@@ -307,3 +327,30 @@ func (e *Engine) serveGRPCWebProxy() {
 		e.log.Err(err).Msg("failed to start the http proxy server")
 	}
 }
+
+// serveREST starts the HTTP REST server
+func (e *Engine) serveREST() {
+
+	e.log.Info().Str("rest_api_address", e.config.RESTListenAddr).Msg("starting REST server on address")
+
+	restAPIHandler := rest.NewRestAPIHandler(e.backend, e.log)
+	e.restServer = rest.NewRestAPIServer(restAPIHandler, e.config.RESTListenAddr, e.log)
+
+	l, err := net.Listen("tcp", e.config.RESTListenAddr)
+	if err != nil {
+		e.log.Err(err).Msg("failed to start the REST server")
+		return
+	}
+
+	e.restAPIAddress = l.Addr()
+
+	e.log.Debug().Str("rest_api_address", e.restAPIAddress.String()).Msg("listening on port")
+
+	err = e.restServer.Serve(l) // blocking call
```

Comment:
Not sure how engines should handle this kind of things but would it make sense to launch as goroutine to not block the thread?

Reply:
yes, it is indeed run in a separate go-routine [here](https://github.com/onflow/flow-go/blob/f8f52fe044566776c3385d065cd689c34f59e465/engine/access/rpc/engine.go#L208)